main/src/library/Iterator.flix

```diff
@@ -14,179 +14,158 @@
  * limitations under the License.
  */
 
-pub enum Iterator[a: Type, r: Region] {
-    case Iterator(Iterator.Done[r], Iterator.Next[a, r])
+pub enum Iterator[a: Type, ef: Bool, r: Region] {
+    case Iterator(Region[r], Unit -> Iterator.Step[a] \ {r, ef})
 }
 
+/*
```

Comment:
Everything in the comment can just be deleted. That's fine.

Reply:
Have done.